src/main/java/com/duckblade/osrs/dpscalc/calc/gearbonus/ObsidianGearBonus.java

```diff
@@ -0,0 +1,79 @@
+package com.duckblade.osrs.dpscalc.calc.gearbonus;
+
+import com.duckblade.osrs.dpscalc.calc.EquipmentItemIdsComputable;
+import com.duckblade.osrs.dpscalc.calc.compute.ComputeContext;
+import com.duckblade.osrs.dpscalc.calc.compute.ComputeInputs;
+import com.duckblade.osrs.dpscalc.calc.model.GearBonuses;
+import com.google.common.collect.ImmutableSet;
+import lombok.RequiredArgsConstructor;
+import net.runelite.api.EquipmentInventorySlot;
+import net.runelite.api.ItemID;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Map;
+import java.util.Set;
+
+@Singleton
+@RequiredArgsConstructor(onConstructor = @__(@Inject))
+public class ObsidianGearBonus implements GearBonusComputable
+{
+	private static final Set<Integer> OBSIDIAN_WEAPONS = ImmutableSet.of(
+		// Is the staff affected? Wiki is unclear. Will anybody ever use it? We'll find out.
+		// ItemID.TOKTZMEJTAL,
+	ItemID.TOKTZXILAK,
+	ItemID.TOKTZXILEK,
+	ItemID.TZHAARKETEM,
+	ItemID.TZHAARKETOM,
+	ItemID.TZHAARKETOM_T
+	);
+
+	private static final Set<Integer> BERSERKER_NECKLACES = ImmutableSet.of(
+		ItemID.BERSERKER_NECKLACE,
+		ItemID.BERSERKER_NECKLACE_OR
+	);
+
+	private static final Set<Integer> OBSIDIAN_HELMETS = ImmutableSet.of(
+		ItemID.OBSIDIAN_HELMET
+	);
+
+	private static final Set<Integer> OBSIDIAN_PLATEBODIES = ImmutableSet.of(
+		ItemID.OBSIDIAN_PLATEBODY
+	);
+
+	private static final Set<Integer> OBSIDIAN_PLATELEGS = ImmutableSet.of(
+		ItemID.OBSIDIAN_PLATELEGS
+	);
+
+	private final EquipmentItemIdsComputable equipmentItemIdsComputable;
+
+	@Override
+	public boolean isApplicable(ComputeContext context)
+	{
+		// Obsidian armour bonuses only apply when attacking i.e. manual casting is forbidden
+		boolean castingSpell = context.get(ComputeInputs.ATTACK_STYLE).isManualCast();
+		return OBSIDIAN_WEAPONS.contains(context.get(equipmentItemIdsComputable).get(EquipmentInventorySlot.WEAPON)) &&
+				!castingSpell;
+	}
+
+	@Override
+	public GearBonuses compute(ComputeContext context)
+	{
+		Map<EquipmentInventorySlot, Integer> equipment = context.get(equipmentItemIdsComputable);
+
+		double preArmourDamage = BERSERKER_NECKLACES.contains(equipment.get(EquipmentInventorySlot.AMULET)) ? 1.2 : 1.0;
+
+		if (OBSIDIAN_HELMETS.contains(equipment.get(EquipmentInventorySlot.HEAD)) &&
+			OBSIDIAN_PLATEBODIES.contains(equipment.get(EquipmentInventorySlot.BODY)) &&
+			OBSIDIAN_PLATELEGS.contains(equipment.get(EquipmentInventorySlot.LEGS)))
+		{
+			// Note: Wiki is unclear if the damage bonuses are additive or multiplicative.
+			// It's a difference between multipliers of 1.3 and 1.32 - difficult to spot.
+			return GearBonuses.of(1.1, 1.1 * preArmourDamage);
```

Comment:
> As long as you change the two bonuses to be in separate classes, this will automatically be done.

Will do,

> although the integer truncation behaviour needs to be changed in GearBonusComputable.

A note on that front. I'm looking at #82 and #78 on one of my local branches.

As you seem to be aware, currently gear bonuses are aggregated naively without any considerations for integer truncation. This is part of the reason for the plugin's inaccuracy as seen in these issues. AggregateGearBonusesComputable will have to go, or otherwise mutate into some Set or whatever.

There's also another problem, wherein the salve amulet and elite void mage set aren't actually applied as a gear bonus for magic damage, it's actually like an additive 20%/2.5% increase (respectively) to the magic damage bonus.

These two otherwise benefit from remaining under the GearBonus structure. However keeping them under there means GearBonuses can no longer remain anonymous as far as whatever new aggregator will be concerned, as these two specific bonuses require special treatment. I was thinking of adding a basic ID field with each GearBonus assigned an integer in a separate file (GearBonusID), but this is pretty horrible and I'm probably missing the forest for the trees, it's an inelegant solution but it would work for resolving the second problem. The slayer helm and tomes still behave like normal GearBonuses for the purposes of magic - it's just the salve amulet / void set that misbehave.

Sorry for the mountains of text, your thoughts? 

PS: I'm fine to keep our conversations here, but I'm open to moving elsewhere if it's more convenient for you.

Reply:
> There's also another problem, wherein the salve amulet and elite void mage set aren't actually applied as a gear bonus for magic damage, it's actually like an additive 20%/2.5% increase (respectively) to the magic damage bonus.

Do you have a source on this? Are all magic damage effects applied as such? (e.g. smoke battlestaff? which seems to be one of the least documented damage bonuses)

> These two otherwise benefit from remaining under the GearBonus structure [...snip]

If they're uniquely applied relative to other GearBonuses, it'll be good to set up a StatsBonus system that modifies the player stats before going into bonuses. The Tumeken's Shadow effect is [implemented poorly](https://github.com/LlemonDuck/dps-calculator/blob/94008b97c6e795e7118bc9d484e050faa4735ffc/src/main/java/com/duckblade/osrs/dpscalc/calc/AttackerItemStatsComputable.java#L39-L47) but would be able to be rolled in properly to that.

> PS: I'm fine to keep our conversations here, but I'm open to moving elsewhere if it's more convenient for you.

Discord is easier (LlemonDuck#3306), but I'll probably forget to reply sometimes there so for discussion like this we can use Discord, and for code that's actually been written let's keep it on GH.

Reply:
> Do you have a source on this? Are all magic damage effects applied as such? (e.g. smoke battlestaff? which seems to be one of the least documented damage bonuses)

My source is the holy grail that everyone seems to agree is correct, it's used by the existing spreadsheets afaik:
https://web.archive.org/web/20190905124128/http://webcache.googleusercontent.com/search?q=cache:http://services.runescape.com/m=forum/forums.ws?317,318,712,65587452

I found it by checking the references on this wiki page:
https://oldschool.runescape.wiki/w/Maximum_magic_hit

Personally in terms of in-game verification, I can't check the smoke battlestaff (ironman btw ;), but I'd be inclined to believe this forum post which states it's a magic damage bonus. The salve amulet however I've seen for myself is certainly a magic damage bonus. Void mage is even more transparent as it shows as a magic damage bonus in the equipment UI in-game. I'm pretty sure as I say that slayer helms and tomes still work with truncation after their effects. 

If you have a bonded account with a smoke battlestaff, you could confirm by casting fire surge with a tormented bracelet, imbued god cape, and smoke battlestaff of course, on a POH dummy. If it's a gear bonus, it will be `int(int(24 * 1.07) * 1.1)` = 27. If it's added to magic bonus, it will be `int(24*1.17)` = 28. 

> If they're uniquely applied relative to other GearBonuses, it'll be good to set up a StatsBonus system that modifies the player stats before going into bonuses. The Tumeken's Shadow effect is [implemented poorly](https://github.com/LlemonDuck/dps-calculator/blob/94008b97c6e795e7118bc9d484e050faa4735ffc/src/main/java/com/duckblade/osrs/dpscalc/calc/AttackerItemStatsComputable.java#L39-L47) but would be able to be rolled in properly to that.

That does sound like it could be more elegant, I'll look into that when I have time.

> Discord is easier (LlemonDuck#3306)

I've sent a friend request (Buddy#6170). I added you a month back and I think you ignored the request which I can't blame you for at all, but check that you haven't also blocked me ;)